cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -1272,6 +1272,30 @@ TEST_F(TestS3FS, FileSystemFromUri) {
   AssertFileInfo(fs.get(), path, FileType::File, 8);
 }
 
+TEST_F(TestS3FS, ConcurrentFSInit) {
```

Comment:
Individual tests are not executed in separate process. The S3 subsystem is probably already initialized by the time this test is executed.

You could instead turn this into a Python test, where it's easier to spawn a subprocess.

Reply:
Yes, I notice that  S3 subsystem actually is set up when S3Environment is set up. I'll make a python test case instead.